### PR TITLE
Show digest of pushed images

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

**Comment by zoidbergwill** [comment]
This looks good to me.

Rebasing may bump some things to reassign who should approve this, since the owners changed in https://github.com/bazelbuild/rules_docker/commit/a4d580ee357b3e40e80becb311ea88214b70e0db

It'd be great to have tests for some of these binaries, but that's definitely out of scope of this PR.

**Review by alexeagle — COMMENTED** [review]
Hi @illicitonion !!

I'm curious what's the motivation for this change? Did you want to copy the digest to paste somewhere? compare the digest with some other reference?

**Comment by illicitonion** [comment]
> Hi @illicitonion !!

👋 🎉 

> I'm curious what's the motivation for this change? Did you want to copy the digest to paste somewhere? compare the digest with some other reference?

Yeah, we have deployment manifests which list the sha256 of the docker images that should be deployed, and so with this change we can trivially copy&paste / scrape them from logs to put them in our deployment manifests. Without this change, we need to look externally for information from the container registry, which it's nice to avoid needing to do, given we already know the information here :)

File: container/go/cmd/pusher/pusher.go [hunk 1]
```diff
@@ -121,11 +121,16 @@ func main() {
 		log.Printf("Destination %s was resolved to %s after stamping.", *dst, stamped)
 	}
 
+	digest, err := img.Digest()
+	if err != nil {
+		log.Fatalf("Failed to digest image: %v", err)
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply container/go/cmd/pusher/pusher.go]: Absolutely - done

**Comment by alexeagle** [comment]
Ah, I think you're just looking for the `.digest`-suffixed output of `container_push` which gives you exactly that info in a file. Then you don't have to scrape the message either - does that work for you?
https://github.com/bazelbuild/rules_docker/blob/master/container/push.bzl#L217-L219

**Comment by illicitonion** [comment]
> Ah, I think you're just looking for the `.digest`-suffixed output of `container_push` which gives you exactly that info in a file. Then you don't have to scrape the message either - does that work for you?
> https://github.com/bazelbuild/rules_docker/blob/master/container/push.bzl#L217-L219

I'm currently using https://github.com/bazelbuild/rules_docker/blob/master/contrib/push-all.bzl which doesn't expose this at the moment. I'm currently just running `bazel run //some:push-all`, and it's not obvious how we'd wire it up to proxy those things through... Any ideas on how we'd go about that?

I guess we'd need to go from one bazel invocation to two, so we could drive building the digest files before/after pushing them?

**Comment by pcj** [comment]
I've also had to resort to extra steps to retrieve the digest myself, so this is a welcome change.

It would be useful IMHO if the printed string was something like `Successfully pushed image to us.gcr.io/foo/ubuntu@sha256:digest`, because then it can be immediately copy-pasted into a kubernetes yaml or docker pull command.

**Review by pcj — APPROVED** [review]
Approving, but would welcome a slightly different format (see commented above)

**Comment by illicitonion** [comment]
Edited to be easier to copy and paste - I didn't want to remove the old text because the label substitution may also be relevant, so I had it do both if the stamped label isn't a digest.

**Comment by gwuah** [comment]
This is a small change that has made my work/life very easy. thanks for this PR @illicitonion

**Pull request merged**